### DISCUSSION TITLE
gcc package does not contain gfortran compiler

### DISCUSSION THREAD

**Discussion by MisesEnForce** [Entware/Entware#1077] [General]
The gcc package provides a C and a C++ compilers but no other compiler (as GNU official gcc does), especially not the fortran compiler gfortran. It would be nice to have it.

I'm ready to test added package on my platform [my use case is synology DS920+] and get a feedback about it.

**Comment by ryzhovau** [comment]
It's a part of the toolchain. We keep it in its current state to avoid any compatibility issues.
Won't add it, sorry

**Comment by MisesEnForce** [comment]
> It's a part of the toolchain. We keep it in its current state to avoid any compatibility issues. Won't add it, sorry

What do you mean with "it's a part of the toolchain" ?

**Comment by ryzhovau** [comment]
Ah, now I get it! You mentioned the native GCC package, not the host GCC compiler for cross-compilation. Sorry. 

We provide native compilation tools [as is](https://github.com/Entware/Entware/wiki/Using-GCC-for-native-compilation) and are not planning to develop them.

**Comment by OlorinOLorien** [comment]
@ryzhovau Hello. I actually have the same "issue" : we have a DS920+ at our company we need to run legacy fortran code on. I understand you won't provide gfortran as openwrt doesn't provide it, fair enough. So I tried to build it from source on the DS920+, using C and C++ compiler still provided by the install of your native gcc package. First I need to build gmp, for which the build went like : 

`wget https://ftp.gnu.org/gnu/gmp/gmp-6.3.0.tar.xz`
`tar -Jxf gmp-6.3.0.tar.xz`
`cd gmp-6.3.0/`
`mkdir build && cd build`
`../configure --prefix=/usr/local/lvm/gmp-6.0.0 --enable-cxx`
`make`

The make yields the error :

`./libtool: line 2642: func__fatal_error: command not found`

That's a huge typo as it should be func_fatal_error instead of func__fatal_error. After correction and remake I finally have the following error, the real serious error :

`libtool:   error: unsupported hardcode properties See the libtool documentation for more information. Fatal configuration error.`

triggered by (the execution of) the command :

`./libtool  --tag=CXX   --mode=link g++  -O2 -pedantic -fomit-frame-pointer -m64 -mtune=k8 -march=k8   -version-info 11:0:7  -o libgmpxx.la -rpath /usr/local/lvm/gmp-6.3.0/lib cxx/dummy.lo cxx/isfuns.lo cxx/ismpf.lo cxx/ismpq.lo cxx/ismpz.lo cxx/ismpznw.lo cxx/limits.lo cxx/osdoprnti.lo cxx/osfuns.lo cxx/osmpf.lo cxx/osmpq.lo cxx/osmpz.lo libgmp.la`

Would you have any idea about what is happening here ?

Other strategy : I could cross build all GCC compilers (gcc, g++, gfortran etc) from my ubuntu 24.04 on WSL2. Do you have any experience in the parameters (arch, platform etc) for Entware running on a DS920+ equipped with an Intel Celeron J4125 ? I naively tried that on my PC : 

`x86_64-linux-gnu-g++ main.cpp -o main`

the main.cpp being

#include <iostream>

`int main()`
`{`
`        std::cout << "Hello, World!\n";`
`        return 0;`
`}`

put the `main` executable on the NAS an ran it and had : 

`./main: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by ./main)`

and I don't any libstdc++ avaibable on entware, so that I would need to cross-build it from source, and for that I would need the kernel source from Entware. Would you have any idea ?
